### ISSUE TITLE
Requests from learners

### ISSUE THREAD

**Issue by omarsar** [dair-ai/ML-Notebooks#14]
- [ ] Add linear regression notebook
- [ ] Add word embeddings notebook

**Comment by Haaris-Rahman** [comment]
Hi,

I could take up the task of creating a linear regression notebook from scratch in pytorch similar to the logistic regression noteboo

**Comment by omarsar** [comment]
@Haaris-Rahman feel free to do so and ping me if you need feedback. Thanks so much for the help.

**Comment by Haaris-Rahman** [comment]
Let me know what you think about the notebook. It is a bit long, but I think it tells a story. 
I would like your feedback on it. I can add the information/text to it as well. But let me know if this is good first step.

It is based on numpy. If you want it, we can change to torch.

**Comment by omarsar** [comment]
hi @Haaris-Rahman took a quick look at the notebook. Are you applying linear regression to a real dataset? If not, something like house prices would be nice. If it makes the notebook too long, then probably two parts might make sense. Thoughts on this?

It's fine if it is Numpy... I could eventually convert it to PyTorch myself.

It's a great start but I think it's hard to follow the notebook in its current state. We hope to make the notebooks as accessible as possible with as much explanations as possible. We can always find all kinds of notebooks online, so we should focus on telling a story, explaining the code bits in a clear way (eg., [here](https://colab.research.google.com/drive/1iBoJ0kngkOthy7SgVaVQA1aHEROt5mra?usp=sharing)), so other learners are encouraged to go through the notebooks and build their knowledge and potentially extend them. Let me know if you need any other feedback.

Again, thank you for contributing. This is looking really nice!

**Comment by Haaris-Rahman** [comment]
Hi @omarsar,

Thank you for the feedback. I will add the explanations next. I wanted to see if you prefer this idea of visualisation or using a real dataset. I will add the real dataset with explanations next.

Do you still want to keep the 2d and 3d dataset part? I thought it would help with visualising what SGD does in different scenarios (overfitting, high learning rate). We could also split that into two.

1. Basic Linear Regression on a real dataset
2. Visualising SGD on Linear Regression in 2D and 3D

**Comment by omarsar** [comment]
That sounds awesome! Yeah, I like that idea a lot.

**Comment by Haaris-Rahman** [comment]
Hi,

I made the updates. Let me know what you think about it.

**Comment by omarsar** [comment]
Looks good to me. I will merge it now. Feel free to add a link to the notebook on the main list. Thanks.
